File: example_assignments/cs110-c/hw08-paycheck_function/solution/paycheck.c
#include <stdio.h>


double calculate_pay(double hours, double rate);


int main() {
  double hours, pay;

  printf("Enter hours worked: ");
  scanf("%lf", &hours);

  printf("Enter the rate of pay: ");
  scanf("%lf", &pay);

  printf("You earned $%.2lf\n", calculate_pay(hours, pay));

  return 0;
}


double calculate_pay(double hours, double rate) {
  double pay;

  if(hours > 40.0) {
    double extra = hours - 41;
    pay = 40 * rate + extra * (1.5 * rate);
  }
  else
    pay = hours * rate;

  return pay;
}

File: example_assignments/cs120-moravian-python/hw16-paycheck_function/solution/paycheck.py
# Calculates the amount of pay for one week
# Parameters:
#   hours - the number of hours worked (positive float)
#   rate  - the normal hourly pay rate (positive float)
# Returns the amount of pay (positive float)
def calculate_pay(hours, rate):
    if hours > 40:
        extra = hours - 40
        pay = 40 * rate + extra * (1.5 * rate)
    else:
        pay = hours * rate

    return pay


hours = eval(input("hours: "))
rate = eval(input("rate: "))

pay = calculate_pay(hours, rate)

print(pay)
